From b2728dc6571f89614ed63e5065b58f8ab38badab Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Wed, 11 Oct 2023 14:45:44 +0200
Subject: [PATCH 0033/1141] serial: stm32: solve typo on "bit and" in
 stm32_usart_iso7816_enabled

This warning has been prompt by LLVM tool chain:
../drivers/tty/serial/stm32-usart.c:467:38: warning: converting the result
of '<<' to a boolean always evaluates to true
[-Wtautological-constant-compare]

Change "logic and" to "bit and" to solve this issue.

Fixes: 6628609f6627 ("serial: stm32: add smart-card protocol")
Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: If678bd7f3993594ebbf6a39144271cac35d43c54
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/333089
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/tty/serial/stm32-usart.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/tty/serial/stm32-usart.c b/drivers/tty/serial/stm32-usart.c
index 9c664dd5db40..47a235cb60ff 100644
--- a/drivers/tty/serial/stm32-usart.c
+++ b/drivers/tty/serial/stm32-usart.c
@@ -464,7 +464,7 @@ static int stm32_usart_config_iso7816(struct uart_port *port, struct serial_iso7
 static bool stm32_usart_iso7816_enabled(struct stm32_port *stm32_port)
 {
 	return stm32_port->port.iso7816.flags &&
-		(stm32_port->port.iso7816.flags && SER_ISO7816_ENABLED);
+		(stm32_port->port.iso7816.flags & SER_ISO7816_ENABLED);
 }
 
 static bool stm32_usart_rx_dma_started(struct stm32_port *stm32_port)
-- 
2.39.2

